### PR TITLE
Enabling rerun of Prow jobs for whitelisted users.

### PR DESCRIPTION
**Pull request by michelle192837**

### CODE REVIEW

**Comment by michelle192837** [comment]
/assign @fejta 
/hold

I'm not 100% on copying deck_rbac.yaml.

**Comment by michelle192837** [comment]
Actually,
/assign @cjwagner 
/unassign @fejta

**Comment by howardjohn** [comment]
fyi @geeknoid I think you wanted something like this.

This will allow us to press the rerun button on https://prow.istio.io right?

<blockquote><div><strong><a href="https://prow.istio.io/">Prow Status</a></strong></div></blockquote>

**Comment by michelle192837** [comment]
Yup! ^^

File: prow/cluster/deck_rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: "deck"
+subjects:
+- kind: ServiceAccount
+  name: "deck"
+  namespace: default
+---
+kind: ClusterRole
```
Comment (cjwagner) [inline prow/cluster/deck_rbac.yaml]: Please remove this ClusterRole and the associated ClusterRoleBinding below. Deck doesn't need this. I'll PR a fix to `k/t-i`.

**Review by cjwagner — CHANGES_REQUESTED** [review]

File: prow/cluster/deck_rbac.yaml (hunk 1 above)
Comment (michelle192837) [reply prow/cluster/deck_rbac.yaml]: Done!

File: prow/config.yaml [hunk 2]
```diff
@@ -40,6 +40,17 @@ deck:
...
+  rerun_auth_config:
+    authorized_users:
+    - chxchx
+    - cjwagner
+    - fejta
+    - hklai
+    - icygalz
+    - rkpagadala
+    - utka
+    - yutongz
```
Comment (cjwagner) [inline prow/config.yaml:53]: + @howardjohn and @geeknoid ?

Currently we just support a list of users who are allowed to trigger any job. Soon we'll also support:
- specifying authorized users per-job in the job config
- allowing anyone who would be able to `/test` or `/retest` a presubmit from GH

Thanks to @mirandachrist for the new functionality!

Comment (cjwagner) [reply prow/config.yaml:53]: Whoops sorry for the delay on this comment Michelle. Feel free to merge as is, we can do this in a follow up.
/lgtm
/approve

Comment (michelle192837) [reply prow/config.yaml:53]: Yeah, I think in a follow up works, others can PR it as well ^^

**Review by cjwagner — APPROVED** [review]

**Comment by michelle192837** [comment]
Argh, didn't remove hold.

/hold cancel

Will need LGTM again.

**Review by howardjohn — APPROVED** [review]

File: prow/config.yaml [hunk 3]
```diff
@@ -40,6 +40,17 @@ deck:
     announcement: "This page -- spyglass -- is the default viewer for prow.{{if .ArtifactPath}} For now, the deprecated gubernator page spyglass replaces <a href='https://gubernator.k8s.io/build/{{.ArtifactPath}}'>remains available</a>.{{end}}"
 
   tide_update_period: 1s
+  # Basing this on istio/test-infra/OWNERS for now.
+  rerun_auth_config:
+    authorized_users:
```
Comment (mirandachrist) [inline prow/config.yaml:45]: This is outdated after https://github.com/kubernetes/test-infra/pull/13564
`authorized_users` will need to become `github_users` after that PR is applied.

**Pull request merged**